### DISCUSSION TITLE
Licensing of example code

### DISCUSSION THREAD

**Discussion by christianmalek** [eclipse-glsp/glsp#1002] [General] [answered]
Hi,

if I use the example code from https://github.com/eclipse-glsp/glsp-examples/tree/master/project-templates/node-json-theia as code base and alter it to work with a specific diagram use case, does this count as "derivative work" in the EPL2.0 license so that I have to disclose my code?

**Comment by planger** [comment, marked as answer]
Hi Christian,

first of all I need to make clear that I'm not a lawyer and can't give any reliable advice on such matters. Thus, I recommend to consult experts in this topic or [contact the Eclipse Foundation](https://www.eclipse.org/org/foundation/contact.php) for such questions.

I think your question falls into the [EPL FAQ 5.4](https://www.eclipse.org/legal/epl-2.0/faq.php#h.xu4ooq8g5mx8).

The intention of the GLSP project is clearly that you can use the project templates as a starting point for your own (potentially proprietary) project. In my humble opinion, in the end what remains equivalent between your code and the project template is purely functional in nature dictated by the technical usage of the GLSP API and therefore not the subject matter for copyright protection. Also, the code of the template is usually only a very small part of the final overall work (di minimis).

However, as I said, I'm not a lawyer and my opinion on this matter isn't worth much. :-)

**Reply by christianmalek** [reply]
Thank you Philip for the quick and comprehensible response.
